src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+                // TODO: add definition context and its highlight
+            }
+
+            // Set location so that Editor can navigate to the result
+            var mappedDocumentSpan = await FindUsagesUtilities.TryMapAndGetFirstAsync(documentSpan, sourceText, cancellationToken).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+
+                result.RelativePath = documentSpan.Document.FilePath;
+                result.LineNumber = location.LinePositionSpan.Start.Line;
+                result.CharacterNumber = location.LinePositionSpan.Start.Character;
+
+                result.PersistentSpan = context.SymbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
+            }
+
+            return result;
+        }
+
+        private static Span AsSpan(TextSpan sourceSpan)
+        {
+            return new Span(sourceSpan.Start, sourceSpan.Length);
+        }
+
+        public string RelativePath { get; private set; }
+
+        public int LineNumber { get; private set; }
+
+        public int CharacterNumber { get; private set; }
+
+        public IPersistentSpan PersistentSpan { get; private set; }
+
+        public ClassifiedTextElement ClassifiedDefinition { get; private set; }
+
+        public Span ContextHighlightSpan { get; private set; }
+
+        public string DocumentGuid { get; private set; }
+
+        public string ProjectGuid { get; private set; }
+
+        public ClassifiedTextElement ContextText { get; private set; }
+
+        public ImageElement Icon { get; private set; }
+
+        public string ProjectName { get; private set; }
+
+        public SymbolSearchResult Definition { get; private set; }
+
+        public bool IsDefinition { get; private set; }
+
+        // Private, because these fields are accessed from TryGetValue until LSP spec is finalized
+
+        private string Kind { get; set; }
+
+        private bool IsReadFrom { get; set; }
+
+        private bool IsWrittenTo { get; set; }
+
+        private string ContainingTypeName { get; set; }
+
+        private string ContainingMemberName { get; set; }
+
+        /// <summary>
+        /// Allows us to pass data whose API hasn't been fully fleshed out in the LSP standard
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public bool TryGetValue(string key, out object value)
+        {
+            switch (key)
+            {
+                case "ContainingTypeName":
```

Comment:
what are these magic constants?

Reply:
This method allows us to pass data whose API hasn't been fully fleshed out in the LSP standard

Reply:
These strings are not part of the published API